GG/src/ListBox.cpp

```diff
@@ -853,20 +853,42 @@ void ListBox::PreRender()
                 NormalizeRow(row);
     }
 
-    AdjustScrolls(false);
+    // Adding/removing scrolls and prerendering rows may change the row sizes and require a change
+    // in added/removed scrolls. This may not be stable. Perform two cycles and if it is not
+    // stable then force the scrollbar to be added if either cycle had a scroll bar.
 
-    // Reset require prerender after call to adjust scrolls
-    Control::PreRender();
+    // Perform a cycle of adjust scrolls and prerendering rows and return if sizes changed.
+    auto check_adjust_scroll_size_change =
+        [this](std::pair<bool, bool> force_scrolls = {false, false})
+        {
+            // This adjust scrolls may add or remove scrolls
```

Comment:
I would de-indent everything from the { to the } line (inclusive). Right now, the actual function body is double-indented from the surrounding scope.